package.json

```diff
@@ -22,7 +22,71 @@
   },
   "license": "MIT",
   "author": "stylelint",
-  "main": "lib/index.js",
+  "type": "module",
+  "exports": {
+    ".": {
+      "import": "./lib/index.mjs",
+      "require": "./lib/index.cjs"
+    },
+    "./package.json": "./package.json",
+    "./lib/utils/isStandardSyntaxAtRule": {
+      "import": "./lib/utils/isStandardSyntaxAtRule.mjs",
+      "require": "./lib/utils/isStandardSyntaxAtRule.cjs"
+    },
+    "./lib/utils/isStandardSyntaxColorFunction": {
+      "import": "./lib/utils/isStandardSyntaxColorFunction.mjs",
+      "require": "./lib/utils/isStandardSyntaxColorFunction.cjs"
+    },
+    "./lib/utils/isStandardSyntaxCombinator": {
+      "import": "./lib/utils/isStandardSyntaxCombinator.mjs",
+      "require": "./lib/utils/isStandardSyntaxCombinator.cjs"
+    },
+    "./lib/utils/isStandardSyntaxComment": {
+      "import": "./lib/utils/isStandardSyntaxComment.mjs",
+      "require": "./lib/utils/isStandardSyntaxComment.cjs"
+    },
+    "./lib/utils/isStandardSyntaxDeclaration": {
+      "import": "./lib/utils/isStandardSyntaxDeclaration.mjs",
+      "require": "./lib/utils/isStandardSyntaxDeclaration.cjs"
+    },
+    "./lib/utils/isStandardSyntaxFunction": {
+      "import": "./lib/utils/isStandardSyntaxFunction.mjs",
+      "require": "./lib/utils/isStandardSyntaxFunction.cjs"
+    },
+    "./lib/utils/isStandardSyntaxHexColor": {
+      "import": "./lib/utils/isStandardSyntaxHexColor.mjs",
+      "require": "./lib/utils/isStandardSyntaxHexColor.cjs"
+    },
+    "./lib/utils/isStandardSyntaxKeyframesName": {
+      "import": "./lib/utils/isStandardSyntaxKeyframesName.mjs",
+      "require": "./lib/utils/isStandardSyntaxKeyframesName.cjs"
+    },
+    "./lib/utils/isStandardSyntaxProperty": {
+      "import": "./lib/utils/isStandardSyntaxProperty.mjs",
+      "require": "./lib/utils/isStandardSyntaxProperty.cjs"
+    },
+    "./lib/utils/isStandardSyntaxRule": {
+      "import": "./lib/utils/isStandardSyntaxRule.mjs",
+      "require": "./lib/utils/isStandardSyntaxRule.cjs"
+    },
+    "./lib/utils/isStandardSyntaxSelector": {
+      "import": "./lib/utils/isStandardSyntaxSelector.mjs",
+      "require": "./lib/utils/isStandardSyntaxSelector.cjs"
+    },
+    "./lib/utils/isStandardSyntaxTypeSelector": {
+      "import": "./lib/utils/isStandardSyntaxTypeSelector.mjs",
+      "require": "./lib/utils/isStandardSyntaxTypeSelector.cjs"
+    },
+    "./lib/utils/isStandardSyntaxUrl": {
+      "import": "./lib/utils/isStandardSyntaxUrl.mjs",
+      "require": "./lib/utils/isStandardSyntaxUrl.cjs"
+    },
+    "./lib/utils/isStandardSyntaxValue": {
+      "import": "./lib/utils/isStandardSyntaxValue.mjs",
+      "require": "./lib/utils/isStandardSyntaxValue.cjs"
+    }
+  },
+  "main": "lib/index.cjs",
```

Comment:
[note] `main` is for backward compatibility.

Reply:
question
nit

There's a risk of revert for this change.
Maybe it would be safer to have it in a separate PR?
i.e. must it be slipstreamed at the same time as l27–30?

Reply:
Make sense. Opened a new PR #7308. 👍🏼